api4/group.go

```diff
@@ -575,23 +575,33 @@ func getGroups(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Err = model.NewAppError("Api4.getGroups", "api.ldap_groups.license_error", nil, "", http.StatusNotImplemented)
 		return
 	}
+	teamID := c.Params.NotAssociatedToTeam
```

Comment:
```suggestion
	var teamID string
	if id := c.Params.NotAssociatedToTeam; model.IsValidId(id) {
	    teamID = id
	}
```

Reply:
Just a suggestion so that you know if these variables are non-default values that they're valid Mattermost IDs formats. 

The way it is currently written you can get down and inside the branches to make the `GetTeam` or `GetChannel` method calls with any non-zero-length string.